packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -226,6 +225,14 @@ class HotRunner extends ResidentRunner {
         device.generator.accept();
       }
       final List<FlutterView> views = await device.vmService.getFlutterViews();
+      final Uri deviceAssetsDirectoryUri = device.devFS.baseUri.resolveUri(globals.fs.path.toUri(getAssetBuildDirectory()));
```

Comment:
This only needs to be performed once at set up - it tells the engine what directory to resolve assets from. Because the engine can load from both the original asset directory and the newly configured directory now, this can be set before any assets are synced